docs/user_guides/fs/feature_view/helper-columns.md

```diff
@@ -0,0 +1,127 @@
+product to assign different weights during the training time.
+
+## Definition
+Both inference and training helper column name(s) must be part of the `Query` object. If helper column name(s) belong to feature group that is part of a `Join` with `prefix` defined, then this prefix needs to prepended
+to the original column name when defining helper column list.
+
+=== "Python"
+
+    !!! example "Define inference and training helper columns with feature views."
+        ```python
+        # define query object 
+        query = label_fg.select("fraud_label")\
+                        .join(cc_profile.select("expiry_date"))\
+                        .join(trans_fg.select(["category", "amount",  "days_until_card_expires", "date_of_transaction"
+                                               "locaton_delta", "longitude", "latitude", "category"])) \
+                        .join(window_aggs_fg.select_except(["trans_volume_mstd", "trans_volume_mavg", "trans_freq", 
+                                                            "loc_delta_mavg"]))
+        
+        # define feature view with helper columns
+        feature_view = fs.get_or_create_feature_view(
+            name='fv_with_helper_col',
+            version=1,
+            query=query,
+            labels=["fraud_label"],
+            transformation_functions=transformation_functions,
+            inference_helper_columns=["longitude", "latitude", "date_of_transaction", "expiry_date" ],
+            training_helper_columns=["category"]
+        )
+        ```
+
+## Retrieval
+When replaying a `Query` during model inference, helper columns will be omitted. However, they can be optionally fetched with inference or training data.
+
+### Batch inference
+
+=== "Python"
+
+    !!! example "Fetch inference helper column values and compute on-demand features for batch inference."
+        ```python
+
+        # import feature functions
+        from feature_functions import location_delta, time_delta
+        
+        # Fetch feature view object  
+        feature_view = fs.get_feature_view(
+            name='fv_with_helper_col',
+            version=1,
+        )
+
+        # Fetch feature data for batch inference with helper columns
+        df = feature_view.get_batch_data(start_time=start_time, end_time=end_time, inference_helpers=True)
+        df['longitude_prev'] = df['longitude'].shift(-1)
+        df['latitute_prev'] = df['latitute'].shift(-1)
+
+        # compute location delta
+        df['loc_delta_t_minus_1'] = df.apply(lambda row: location_delta(row['longitude'], 
+                                                                        row['latitute'],
+                                                                        row['longitude_prev'], 
+                                                                        row['latitute_prev']), axis=1)
+
+        # compute time delta
+        df['days_until_card_expires'] = df.apply(lambda row: time_delta(row['date_of_transaction'],  
```

Comment:
Same comment as above, have a single example so that what we care about doesn't get lost in all this code.